projects/packages/jetpack-mu-wpcom/tests/php/features/launchpad/class-launchpad-task-lists-test.php

```diff
@@ -22,4 +22,86 @@ public function test_build_creates_no_PHP_warnings() {
 		$this->assertIsArray( $result );
 		$this->assertEmpty( $result );
 	}
+
+	/**
+	 * Callback to check if the first task in a task list is completed.
+	 *
+	 * @param array $task_list Task list.
+	 */
+	public function callback_first_task_completed( $task_list ) {
+		$first_task = reset( wpcom_launchpad_checklists()->build( $task_list['id'] ) );
+		return $first_task['completed'];
+	}
+
+	/**
+	 * Data provider for ::test_task_list_is_completed().
+	 */
+	public function provider_test_task_list_is_completed() {
+		$incomplete_task = array();
+		$completed_task  = array(
+			'is_complete_callback' => '__return_true',
+		);
+
+		$launch_task = array(
+			'isLaunchTask' => true,
+		);
+
+		$launch_task_completed = array(
+			'isLaunchTask'         => true,
+			'is_complete_callback' => '__return_true',
+		);
+
+		return array(
+			// tasks, expected, task_list_options
+			array( array( $incomplete_task, $incomplete_task ), false ),
+			array( array( $completed_task, $completed_task ), true ),
+			array( array( $incomplete_task, $launch_task ), false ),
+			array( array( $incomplete_task, $launch_task_completed ), true ),
+			array( array( $incomplete_task, $completed_task ), true, array( 'require_last_task_completion' => true ) ),
+			array( array( $completed_task, $incomplete_task ), false, array( 'require_last_task_completion' => true ) ),
+			array( array( $completed_task, $incomplete_task ), true, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			array( array( $incomplete_task, $incomplete_task ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			array( array( $incomplete_task, $completed_task ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			array( array( $incomplete_task, $launch_task_completed ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			array( array( $incomplete_task, $incomplete_task ), true, array( 'is_completed_callback' => '__return_true' ) ),
+			array( array( $completed_task, $incomplete_task ), true, array( 'is_completed_callback' => array( $this, 'callback_first_task_completed' ) ) ),
+		);
```

Comment:
It would be really helpful to describe each test case. I've suggested a few keys, but not all.
```suggestion
		return array(
			// tasks, expected, task_list_options
			'all tasks incomplete should be incomplete using default options' => array( array( $incomplete_task, $incomplete_task ), false ),
			'all tasks complete should be complete using default options' => array( array( $completed_task, $completed_task ), true ),
			'incomplete task and incomplete launch task should be incomplete using default options' => array( array( $incomplete_task, $launch_task ), false ),
			array( array( $incomplete_task, $launch_task_completed ), true ),
			array( array( $incomplete_task, $completed_task ), true, array( 'require_last_task_completion' => true ) ),
			array( array( $completed_task, $incomplete_task ), false, array( 'require_last_task_completion' => true ) ),
			array( array( $completed_task, $incomplete_task ), true, array( 'required_task_ids' => array( 'task_0' ) ) ),
			array( array( $incomplete_task, $incomplete_task ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
			array( array( $incomplete_task, $completed_task ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
			array( array( $incomplete_task, $launch_task_completed ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
			array( array( $incomplete_task, $incomplete_task ), true, array( 'is_completed_callback' => '__return_true' ) ),
			array( array( $completed_task, $incomplete_task ), true, array( 'is_completed_callback' => array( $this, 'callback_first_task_completed' ) ) ),
		);
```